snuba/clickhouse/query.py

```diff
@@ -36,10 +34,17 @@ def format(self) -> str:
         select_clause = u'SELECT {}'.format(', '.join(group_exprs + aggregate_exprs + selected_cols))
 
         from_clause = u'FROM {}'.format(source)
-        if self.__final:
+
+        if self.__request.query.get_final():
             from_clause = u'{} FINAL'.format(from_clause)
-        if query.get_sample():
-            from_clause = u'{} SAMPLE {}'.format(from_clause, query.get_sample())
+
+        if self.__request.settings.turbo:
+            sample_rate = settings.TURBO_SAMPLE_RATE
```

Comment:
This does not seem right.
The previous implementation was :
```
if turbo:
      if request.query.get_sample() is None:	
            request.query.set_sample(settings.TURBO_SAMPLE_RATE)
```

This one does not apply the query provided sample rate if turbo is true.